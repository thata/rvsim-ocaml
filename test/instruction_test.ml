let () = print_endline "Hi"

(* メモリを初期化 *)
let memory = Rvsim.Memory.of_string "\x93\x00\x01\x01\x94\x00\x01\x01"

(* アドレスを指定して読み込み *)
let test = Rvsim.Memory.read_word memory 0
let _ = Printf.printf "%08x\n" (Int32.to_int test)
let test = Rvsim.Memory.read_word memory 4
let _ = Printf.printf "%08x\n" (Int32.to_int test)

(* アドレス0番地へ書き込み *)
let test =
  Rvsim.Memory.write_word memory 0 0x00ABCDEFl;
  Rvsim.Memory.read_word memory 0

let _ = Printf.printf "%08x\n" (Int32.to_int test)

(* アドレス4番地へ書き込み *)
let test =
  Rvsim.Memory.write_word memory 4 (-1l);
  Rvsim.Memory.read_word memory 4

let _ = Printf.printf "%d\n" (Int32.to_int test)
