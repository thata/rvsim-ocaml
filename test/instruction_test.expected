51
17
5
19
23
65
19
17
5
19
2049
35
5
19
23
2097
99
5
19
23
7218
